src/fde.h

```diff
@@ -84,8 +84,25 @@ class fde
     /// use I/O methods that maintain an internal-to-them buffer
     void useBufferedIo(READ_HANDLER *, WRITE_HANDLER *);
 
-    int read(int fd, char *buf, int len) { return readMethod_(fd, buf, len); }
-    int write(int fd, const char *buf, int len) { return writeMethod_(fd, buf, len); }
+    /// call the read method set by fde::setIo() requesting len more bytes.
+    /// Performs I/O counting for all bytes received.
+    /// \returns the number of bytes actually received
+    int read(int fd, char *buf, int len) {
+        auto n = readMethod_(fd, buf, len);
+        if (n > 0)
+            bytesRead(n);
+        return n;
+    }
+
+    /// call the WRITE_HANDLER set by fde::setIo() with len more bytes.
+    /// Performs I/O counting for all bytes written.
+    /// \returns the number of bytes actually written
+    int write(int fd, const char *buf, int len) {
```

Comment:
Please apply read() changes to the write() method documentation as well.

Making `n` (and other variables) constant would be nice.

Reply:
done in [ce5f427](https://github.com/squid-cache/squid/pull/1352/commits/ce5f4270e3dbfeba2771540abf1ff2e46f2bd02d)